### PR TITLE
Snow 1065143 fix FQN support for streamlit

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add support for fully qualified name as streamlit name

### CODE REVIEW

File: src/snowflake/cli/plugins/streamlit/commands.py [hunk 1]
```diff
@@ -89,9 +89,21 @@ def streamlit_deploy(
...
     """
     streamlit = cli_context.project_definition
+
     if not streamlit:
         return MessageResult("No streamlit were specified in project definition.")
 
+    database = streamlit.get("database")
+    schema = streamlit.get("schema")
+    options.pop("database")
+    options.pop("schema")
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/streamlit/commands.py:99]:
```suggestion
    database = options.pop("database", None)
    schema = options.pop("schema", None)
```

Will do the same, but why we need pop?

File: src/snowflake/cli/plugins/streamlit/commands.py [hunk 2]
```diff
@@ -89,9 +89,21 @@ def streamlit_deploy(
...
+
+    number_of_fqn_parts_in_name = len(streamlit["name"].split("."))
+    if number_of_fqn_parts_in_name >= 3 and database:
+        raise ClickException(f"database of the streamlit is redefined in its name")
+    if number_of_fqn_parts_in_name >= 2 and schema:
+        raise ClickException(f"schema of the streamlit is redefined in its name")
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/streamlit/commands.py:105]: Should we make this a common logic?

File: src/snowflake/cli/plugins/streamlit/manager.py [hunk 3]
```diff
@@ -99,6 +101,13 @@ def deploy(
         **options,
     ):
         stage_manager = StageManager()
+        # for backwards compatibility - quoted stage path might be case-sensitive
+        # https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers
+        streamlit_name_for_root_location = streamlit_name.split(".")[-1]
+        fully_qualified_name = stage_manager.to_fully_qualified_name(
+            streamlit_name, database=database, schema=schema
+        )
+        streamlit_name = fully_qualified_name.split(".")[-1]
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/streamlit/manager.py:110]: Introduce similar method as `get_stage_name_from_path` (`get_object_name_from_fqn`)

File: src/snowflake/cli/plugins/streamlit/commands.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/streamlit/commands.py:99]: "options" do not seem to catch "database" and "schema" from `snowflake.yml`.
We need "pop" as later we gat duplicated keyword arguments:
```
    url = StreamlitManager().deploy(
        streamlit_name=streamlit["name"],
        database=database,
        schema=schema,
        ...
        **options,
    )
```

Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/plugins/streamlit/commands.py:99]: Oh, I a see. Do we still need to pop from options?

Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/streamlit/commands.py:99]: An alternative would be
```py
options['database'] = streamlit.get('database') or options['database']
```
options are later passed to `StreamlitManager().deploy()` as kwargs, and already have 'database' and 'schema' keys inside. I decided to pass them explicitly to increase readability, WDYT?

File: src/snowflake/cli/plugins/streamlit/commands.py (hunk 2 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/streamlit/commands.py]: done

File: src/snowflake/cli/plugins/streamlit/manager.py (hunk 3 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/streamlit/manager.py]: done

File: src/snowflake/cli/plugins/streamlit/commands.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/streamlit/commands.py]: Discussed on slack - parameters will be moved as separate PR

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -3,6 +3,8 @@
 ## Backward incompatibility
 
 ## New additions
+* Added possibility to specify `database` and `schema` for streamlit apps by providing fully qualified
+  name in `snowflake.yml` (`name = database.schema.name`)
```
Comment (sfc-gh-turbaszek) [inline RELEASE-NOTES.md:7]: Should we adjust the note?

Comment (sfc-gh-pczajka) [inline RELEASE-NOTES.md:7]:
```suggestion
* Added support for fully qualified name (`database.schema.name`) in `name` parameter in streamlit project definition
```

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**